sdk/cosmosdb/cosmos/src/queryExecutionContext/defaultQueryExecutionContext.ts

```diff
@@ -216,6 +232,21 @@ export class DefaultQueryExecutionContext implements ExecutionContext {
           responseHeaders[Constants.HttpHeaders.QueryMetrics]["0"] = queryMetrics;
         }
 
+        if (
+          operationOptions &&
+          operationOptions.ruCapPerOperation &&
+          ruConsumed &&
+          ruConsumed.value !== undefined
+        ) {
+          ruConsumed.value += getRequestChargeIfAny(responseHeaders);
+          if (ruConsumed.value > operationOptions.ruCapPerOperation) {
+            // For RUCapPerOperationExceededError error, we will be marking state as inProgress as we want to support continue
```

Comment:
Thanks for the comments.

Reply:
Where are we doing this?

Reply:
This situation occurs in the current implementation because, following the RUCapPerOperationExceededError, we do not modify the state; instead, we maintain it in the same state as before, as per the previous implementation.